### DISCUSSION TITLE
HidHIDE remove failed

### DISCUSSION THREAD

**Discussion by CorviGaming** [nefarius/HidHide#66] [Q&A]
Hi, 
my issu is with the named program in title.
In windows 11 x64 enviroment.
I was planning to use hidhide to hid me ps4 controller but in the process the files get corrupted. 
Can not delete from regedit or from drivers list via Driver store explorer.
Somehow i have to version of the driver (prntscr attached). None of them is uninstallable. 
And the controll panel not foundable. 

Can u help me remove the drivers?

Sincerly,
Corv
![drivers](https://user-images.githubusercontent.com/101871947/158996695-5e6836d6-689d-494a-ad12-2399169cf58d.png)
![Error](https://user-images.githubusercontent.com/101871947/158996698-11440e6d-4648-4b47-9b5b-2a0683122bfa.png)
i

**Comment by nefarius** [comment]
**Never ever use Driver Store Explorer to remove a filter driver**, you can brick your machine!

Besides the official uninstaller you can use [this tool instead](https://github.com/nefarius/Legacinator).

Cheers

**Reply by CorviGaming** [reply]
Hey Nefarius, 
I did what u asked. 
Dindt worked. 
![kép_2022-03-18_132530](https://user-images.githubusercontent.com/101871947/159002601-20d2ffc8-419b-40d8-8799-f176c5ba636d.png)

**Reply by nefarius** [reply]
Reinstall HH using the official setup and attempt an uninstall afterwards.

**Reply by CorviGaming** [reply]
I did what u asked: https://www.youtube.com/watch?v=OgraOw7_fv0&list=PLRVnKuiaLeYneRpqT8MnrviI5EPsWIKzh

now the option disapiared from the ds4 and still cant find the app in the apps.

**Reply by nefarius** [reply]
Link not working.

**Reply by nefarius** [reply]
WDYM "the option disappeared", I don't understand.

**Comment by 0lm** [comment]
Hello. I hope you don't mind me reviving this topic. I currently have the same issue. I uninstalled HidHide in the Softwarelist, but DriverExplorer gave me an error. Now I am reading "Never ever use Driver Store Explorer to remove a filter driver, you can brick your machine!" - what exactly does this mean? 

Because after uninstalling old stuff, I already used Driverexplorer to remove vigembus.inf, fireshock.inf and a really old scpvbus.inf (i removed those via Dirverexplorer, after properly ending the services via cmd and uninstalling the softwares in the software/app list in Windows). Was this right, or did these also count as "filter drivers"? (Reason why I even used DriverExplorer in the first place, was, because it was recommended by you [here](https://github.com/nefarius/ViGEmBus/issues/48) back then for removing traces and I thought after properly uninstalling HidHide, the hidhide.inf would be also just a trace.)

So, now I did as yousaid and tried Legacinator first, but it couldn't find any traces of old drivers. Then I reinstalled HidHide sing the current installer. But to my (shocking?) surprise, it installed it to a differnt location. 

The old Hidhide 1.0.18 is installed here:
C:\Windows\System32\DriverStore\FileRepository\hidhide.inf_amd64_1b0290...

While the newest reinstalled Hidhide 1.4.181 is installed here:
C:\Windows\System32\DriverStore\FileRepository\hidhide.inf_amd64_7c068d...

They are both handled differently as it seems. The older 1.0.18 is never overwritten by the newer install.

**Reply by nefarius** [reply]
Use the official setup provided uninstaller, end of story. Using DSE on filter drivers can brick your machine, it's all documented in my docs site, you can't just use DSE on any driver type without knowing what you're doing. I never recommended using DSE on HidHide, you can't just mix different drivers and projects together and assume it's all the same, it is not.

**Reply by nefarius** [reply]
Ofc. they are different, the driver store keeps a copy of each unique version, it has nothing to do with whatever is the currently used active driver. If you do not know these details, I recommend not mucking around with it.

**Reply by 0lm** [reply]
Thanks for the reply. That's why I asked here, to be sure, since I assumed DSE is the solution for any old traces (after successfull uninstalls). Well, good thing I got an error with DSE when removing hidhide.inf then. I also edited my comment a bit, to show you a new issue with Hidhide not overwriting an older version. Still, going to restarted now and do test if Legacinator maybe will find the older versio,n too, now.

EDIT: The thing is, I now and then (like once every few months) get USB issues, where the USB Ports stop working and only restarting the PC make them work again. That's why I try to remove HidHide 1.0.18.

**Reply by nefarius** [reply]
I do not recall from memory if the uninstaller cleans out the driver store since that doesn't really affect anything anyway except wasting a few Kilobytes of your disk space.

**Reply by nefarius** [reply]
Drivers sitting in the driver store doesn't mean anything about in use drivers, making that assumption is misguided. If HidHide is not loaded as a class filter, it's not doing anything, driver store copy or not, it doesn't matter one bit.

**Reply by 0lm** [reply]
Thanks, I guess I will leave it like this, then, if my last try also won't work.

By now, I tested something: I installed the newest HidHide and then uninstalled it via the Widows10-Layout App List, it prompted me a Restart. After restart, the newest hidhide.inf also disappeared (DSE only showed 10.018 and not two different ones anymore) But I remember when I uninstalled hidhide 1.018 (i used the old Windows 7 Software uninstall layout that time btw.) I was not prompted a restart. Next I will try to reinstall hidhide 10.0.18 and try uninstalling again but this time via the Windows10-App-layout list. (basically, my last try now)

**Reply by nefarius** [reply]
I don't recall what is going on that many versions ago, read the changelogs and commits since then if you wanna see exactly what changed over time. In summary: use the provided setup and all will be fine, the end.

**Reply by 0lm** [reply]
I came to a conclusion. Looks like I am stuck with hidhide 1.0.18 driver store drivers. Tried to reinstall Hidhide 1.0.18 (apparently it is the very first release of hidhide) and i got an error, saying installation ended prematurely. I assume it is because of hidhide.inf 1.0.18 still existing.

Would you say it is safe to remove it in Windows Save Mode? I used ProcessExplorer if somehing is using hidhide.inf but there were no results. The Hidhide service is also not running (since it should be uninstalled), so, I have no idea what is preventing 1.0.18 from getting deleted.

**Reply by nefarius** [reply]
Why are you so obsessed with that driver store copy, it doesn't do anything, it doesn't matter, leave it be 😅 if even DSE can't remove it, it is probably damaged in a way both of us do not know how to repair anyway. Move on, mate.

**Reply by 0lm** [reply]
Well I guess you are right. But one last thing, maybe it helps you to understand what is going on (because I dont understand, lol): I tried to manually rename/move away/delete hidhide.inf from its path (  C:\Windows\System32\DriverStore\FileRepository\hidhide.inf_amd64_1b0290... ) and I got the following popup: "You need SYSTEM rights to do this". My highest rights I can get on my Windows 10 Useraccount are ADMIN. Could it be because of right-issues with 1.0.18, it couldn't be removed when I uninstalled the App?

**Reply by nefarius** [reply]
That directory is protected from goofs like you that manually try to poke around in it for good reasons 😛 if you can't figure out how and why, you have no business meddling with it to begin with. If you value your working PC, educate yourself or leave it alone. This has nothing to do with why an incredibly outdated setup doesn't work on your machine. Good luck!

**Reply by 0lm** [reply]
Well, I guess it is a good way to prevent unexperienced users to access the files there. I can reassure you, I am a little bit more experienced and know that the Driver Store is a really risky and sensitive path. The thing is, if it was possible to install a driver, which did not exist before, then it has to be possible to safely and properly remove it again. It is not like hidhide 10.018 was a driver installed by Windows 10 itself during the OS installation.

I now managed to remove the leftovers of hidhide. The solution was actually simple: DSE remove driver, but without forcing it. Looks like if you force to remove hidhide (1.0.18 specifically), then it will give you an error. After removing it, I was able to reinstall Hidhide 1.0.18 again properly without any error. Then I tried to uninstall it again via Winsows10-Layout App-List and it uninstalled properly. But it still did not remove the driver store leftovers again.

I think the first release of Hidhide (which is v.1.0.18) just wasn't configured to remove the leftover traces, that's all. And for some reason, DSE won't remove that leftover either, if you selected "force deletion". But deleting normally with the "force deletion" unticked, it worked.

But as you suggested, if the Driver isn't doing any harm and is not loaded anymore by anything, then just don't touch it. In this case, I just was sure it should be removeable because it was a 3rd party driver and its software, which usually loads it, was properly uninstalled before.

So, if anyone should encounter the same issue with v1.0.18: Uninstall HidHide normally from the Windows App List. Check your Services in Task manager and in your System Service list (services.msc) to make sure, that hidhide is not running anymore, or even stop that service via cmd command. Maybe also restart (because v1.0.18 uninstall of hihide software won't promot a restart). After restart, check again your services to be sure hidhide is not loaded anymore. Only after uninstalling and making sure hidhide is not used anymore by anything, then it should be safe to remove the leftovers via DriverStoreExplorer, but **without** "force deletion". For some reason, if you do it with "force deletion", it will give you an error. - Still, have in mind: This worked for me. I can't compare to other cases since I don't know any. Maybe I was just lucky (although I tested this a few times, by reinstalling hihide v1.0.18 again to be sure). So yeah, listen to Nefarius and don't touch the Driver Store. Uninstalling the Hidhide v1.0.18 software in the App list should be totally enough to make Windows never touch hidhide.inf anymore. Only reason I presented my solution here, was: If you are like me and small thing slike that are bugging you.

**Reply by nefarius** [reply]
You don't need to assure **me** of anything, it's your PC, your accountability, be my guest 😛

As I said previously, 1.0.18 was a long time ago, who knows what the uninstaller did or did not do.

Drivers are not installed under the current users account so it is not surprising at all that different, more restricted folder structures exist that even the Admin has no business in accessing, that's been a thing since Vista. 

If you'd have simply installed and uninstalled the latest setup, you'd have the same effect. That 1.0.18 copy in the driver store did nothing except allocate a bit of space and is not related to your USB problems whatsoever since HidHide is not a USB driver. 

Your desire to have a clean system is admirable but IMHO not worth this effort but you do you 👍

Cheers

**Reply by 0lm** [reply]
Yup, I noticed that. It wouldn't have done any harm if I just let that leftover hidhide.inf untouched, since as you said, uninstalling Hidhide Software from App list is totally enough. It was just a me-thing, that I had to find out why it couldn't be removed, lol. Thank you for your time, for guiding me and trying to help!
